pypose/module/imu_preintegrator.py

```diff
@@ -8,34 +8,47 @@ class IMUPreintegrator(nn.Module):
-        acc_cov (float, optional): covariance of the accelerator. Default: (2e-3)**2
-        prop_cov (Bool, optional): flag to propogate the covariance matrix. Default: :obj:`True`
+        gyro_cov: covariance of the gyroscope. If the covariance of the x, y and z axises are
+            different, the :obj:`gyro_cov` use the form :code:`torch.Tensor(gyro_cov_x, gyro_cov_y,
+            gyro_cov_z)`. Default: (3.2e-3)**2
+        acc_cov: covariance of the accelerator. If the covariance of the x, y and z axises are
+            different, the :obj:`acc_cov` use the form :code:`torch.Tensor(acc_cov_x, acc_cov_y,
+            acc_cov_z)`. Default: (8e-2)**2
+        prop_cov (Bool, optional): flag to propagate the covariance matrix. Default: :obj:`True`
         reset (Bool, optional): flag to reset the initial states after each time the :obj:`forward`
             function is called. If False, the IMU integrator will use the states from last time
-            as the initial states. Note that if the :obj:`init_state` is not :obj:`None` while calling 
-            the :obj:`forward` function, the integrator will use the given inital state Default: :obj:`False`.
+            as the initial states. This flag is ignored if the :obj:`init_state` is not ``None``,
+            the integrator will use the given initial state. Default: :obj:`False`.
     '''
     def __init__(self, pos = torch.zeros(3),
                        rot = pp.identity_SO3(),
                        vel = torch.zeros(3),
                        gravity = 9.81007,
-                       gyro_cov = (1.6968e-4)**2,
-                       acc_cov = (2e-3)**2,
+                       gyro_cov = (3.2e-3)**2,
+                       acc_cov = (8e-2)**2,
                        prop_cov = True,
                        reset = False):
         super().__init__()
-        self.reset, self.prop_cov, self.gyro_cov, self.acc_cov = reset, prop_cov, gyro_cov, acc_cov
+        self.reset, self.prop_cov = reset, prop_cov
+
+        if isinstance(acc_cov, float):
+            acc_cov = torch.tensor([[acc_cov, acc_cov, acc_cov]])
+        if isinstance(gyro_cov, float):
+            gyro_cov = torch.tensor([[gyro_cov, gyro_cov, gyro_cov]])
+        
         # Initial status of IMU: (pos)ition, (rot)ation, (vel)ocity, (cov)ariance
         self.register_buffer('gravity', torch.tensor([0, 0, gravity]), persistent=False)
         self.register_buffer('pos', self._check(pos).clone(), persistent=False)
         self.register_buffer('rot', self._check(rot).clone(), persistent=False)
         self.register_buffer('vel', self._check(vel).clone(), persistent=False)
         self.register_buffer('cov', torch.zeros(1, 9, 9), persistent=False)
+        self.register_buffer('gyro_cov', gyro_cov, persistent=False)
+        self.register_buffer('acc_cov', acc_cov, persistent=False)
+        self.Rij = None # th  referenced rotation of the covariance
```

Comment:
comments of this line are unclear.

Reply:
Do you have to add `self.Rij` here?

Reply:
Yes, with the flag of `self.reset`, if you want to track the correct covariance with a non-zero `init_cov`, you need to maintain both the `self.cov` and `self.Rij`.